package/MDAnalysis/core/groups.py

```diff
@@ -388,6 +388,37 @@ def __getattr__(self, attr):
                 err += 'Did you mean {match}?'.format(match=match)
             raise AttributeError(err)
 
+    def get_connections(self, typename, outside=False):
+        """
+        Get bonded connections between atoms as a
+        :class:`~MDAnalysis.core.topologyobjects.TopologyGroup`.
+
+        Parameters
+        ----------
+        typename : str
+            group name. One of {"bonds", "angles", "dihedrals",
+            "impropers", "ureybradleys", "cmaps"}
+        outside : bool (optional)
+            Whether to include connections involving atoms outside
+            this group.
+
+        Returns
+        -------
+        TopologyGroup
+            containing the bonded group of choice, i.e. bonds, angles,
+            dihedrals, impropers, ureybradleys or cmaps.
+
+        .. versionadded:: 1.1.0
+        """
+        # AtomGroup has handy error messages for missing attributes
+        ugroup = getattr(self.universe.atoms, typename)
```

Comment:
So we're not using `atomgroup_intersection` here because this method can handle not just things in a `TopologyGroup`?

Reply:
I think it was more to avoid having to distinguish between Atoms and AtomGroups (etc) with ``ix_array``, as ``atomgroup_intersection`` only accepts AtomGroups

Reply:
I just realised that `Residue.bonds` is not existing functionality, so I added tests.